lib/Parser/Parse.cpp

```diff
@@ -1701,6 +1703,12 @@ void Parser::BindPidRefsInScope(IdentPtr pid, Symbol *sym, int blockId, uint max
             sym->SetIsModuleExportStorage(true);
         }
 
+        if (ref->GetFuncScopeId() != funcId && !sym->GetIsGlobal())
+        {
+            Assert(ref->GetFuncScopeId() > funcId);
```

Comment:
So this means that the ref here is coming from a nested function, correct?

I'm not following what this is always the case, versus a ref coming from an enclosing scope.  I will admit my understanding of the PidRef model is not good.